pwndbg/heap/ptmalloc.py

```diff
@@ -1191,14 +1191,55 @@ def bin_at(self, index, arena_addr=None):
         chain_fd = get_chain(front, fd_offset)
         chain_bk = get_chain(back, bk_offset)
 
-        # check if bin[index] points to itself (is empty)
-        if len(chain_fd) == len(chain_bk) == 2 and chain_fd[0] == chain_bk[0]:
-            chain_fd = [0]
-            chain_bk = [0]
-
-        # check if corrupted
-        elif chain_fd[:-1] != chain_bk[:-2][::-1] + [chain_bk[-2]]:
+        if len(chain_fd) != len(chain_bk):
+            # If the chain lengths aren't equal, the chain is corrupted
             is_chain_corrupted = True
+        elif len(chain_fd) < 2 or len(chain_bk) < 2:
+            # Chains containing less than two entries are corrupted, as the smallest
+            # chain (an empty bin) would look like something like `[main_arena+88, 0]`.
+            is_corrupted = True
```

Comment:
I may be missing something, but I don't see this variable being used anywhere.

Reply:
Sorry, I meant `is_chain_corrupted` here.

Reply:
Yeah I guess there must be a gap in the current tests since this still passed even with an unused variable.
Probably worth a adding a test that checks for this case wouldn't you say?